clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java

```diff
@@ -61,18 +61,26 @@ public void startConsumption(VeniceStoreVersionConfig storeConfig, int partition
         Utils.waitStoreVersionOrThrow(storeVersion, getStoreIngestionService().getMetadataRepo());
     Supplier<StoreVersionState> svsSupplier = () -> storageMetadataService.getStoreVersionState(storeVersion);
     syncStoreVersionConfig(storeAndVersion.getFirst(), storeConfig);
-    AbstractStorageEngine storageEngine = storageService.openStoreForNewPartition(storeConfig, partition, svsSupplier);
-    topicStorageEngineReferenceMap.compute(storeVersion, (key, storageEngineAtomicReference) -> {
-      if (storageEngineAtomicReference != null) {
-        storageEngineAtomicReference.set(storageEngine);
-      }
-      return storageEngineAtomicReference;
-    });
 
+    long startTime = System.currentTimeMillis();
     CompletionStage<Void> bootstrapFuture =
         bootstrapFromBlobs(storeAndVersion.getFirst(), storeAndVersion.getSecond().getNumber(), partition);
 
     bootstrapFuture.whenComplete((result, throwable) -> {
+      long endTime = System.currentTimeMillis();
+      LOGGER.info(
```

Comment:
OK I know why this happens basically for the II + memory limiter integration test...

With current change, the call will happen within an async thread, when the error in memory limiter is thrown, the error is not caught but the push job is stuck, so the test eventually times out.

I think these two features are not major Da Vinci features and are not related to your feature.
In order to help you to unblock, I think this works:

use a Runnable to capture all the action we want to do here inside the whenComplete.
Move the blob transfer check:
 if (!storeAndVersion.getFirst().isBlobTransferEnabled() || storeAndVersion.getFirst().isHybrid() || blobTransferManager == null) 
from the below method to here,
when not doing blob transfer, directly call runnable.run(), otherwise, call runnable.run() in the whenComplete part.

It is not a prefect solution, but I think it is good enough to unblock here....

Reply:
launched in latest commit, please review it when you get a chance